### PR TITLE
Feature/#54

### PR DESCRIPTION
**Pull request by BerendWouters**
This feature adds a new form when:
- creating a new transaction and selecting the membership fee
- editing a transaction and changing the membership fee, member or the amount of the membership fee

The form calculates the amount of months will be added to the current expiration date, based on the amount of the membership fee.

Additionally, we set the expiration date to the end of the month (when it is set).

What could be done better?
It's now required to change the expiration date explicitly, by clicking on the button. It would be better/easier to just change it by clicking on the "Add" or "Save" button at the button of the page.

### CODE REVIEW

File: Web/Pages/Transactions/EditTransaction.razor [hunk 1]
```diff
@@ -81,5 +83,24 @@
...
+        var memberId = Transaction.Counterparty.MemberId.Value;
+        var member = await _mediator.Send(new GetMemberByIdQuery(memberId));
+        var transactionTypeForMembershipFee =
+            Transaction.TransactionTypeAmounts.SingleOrDefault(x => x.TransactionType == TransactionType.DebitMemberFee);
+        if (transactionTypeForMembershipFee != null && transactionTypeForMembershipFee.Amount != 0)
+        {
+            var amountOfMonths = (int)Math.Floor((double)transactionTypeForMembershipFee.Amount / member.MembershipFee);
+            if (member.MembershipExpiryDate.HasValue)
+            {
+                //Transaction.NewExpirationDate = member.MembershipExpiryDate.Value.AddMonths(amountOfMonths).Date;
```
Comment (vyruz1986) [inline Web/Pages/Transactions/EditTransaction.razor:101]: This shouldn't be commented out I think?

File: Web/Pages/Transactions/EditTransaction.razor [hunk 2]
```diff
@@ -81,5 +83,24 @@
         _snackbar.Clear();
         _snackbar.Add("Please correct all errors on the form", Severity.Error);
     }
-
+    private async Task ChangeMembershipFee()
```
Comment (vyruz1986) [inline Web/Pages/Transactions/EditTransaction.razor:86]: This doesn't seem to be called anywhere?

**Review by vyruz1986 — CHANGES_REQUESTED** [review]
I don't like how this is very UI-driven. I would rather see a `MembershipFeePaid` event being published, which is then picked up by a `ExtendMembershipHandler` handler which would extend the expiry date accordingly.
This would make for a more flexible system, as we could in the future have e.g. a `ThankMemberForPayingMembershipFee` handler which sends an email to the member thanking them for paying their fee.

**Comment by BerendWouters** [comment]
> I don't like how this is very UI-driven. I would rather see a `MembershipFeePaid` event being published, which is then picked up by a `ExtendMembershipHandler` handler which would extend the expiry date accordingly. This would make for a more flexible system, as we could in the future have e.g. a `ThankMemberForPayingMembershipFee` handler which sends an email to the member thanking them for paying their fee.

I'm not really familiar with events in "The Grand Scheme of Things Called DDD", so I'll have to look into that.

**Comment by vyruz1986** [comment]
> > I don't like how this is very UI-driven. I would rather see a `MembershipFeePaid` event being published, which is then picked up by a `ExtendMembershipHandler` handler which would extend the expiry date accordingly. This would make for a more flexible system, as we could in the future have e.g. a `ThankMemberForPayingMembershipFee` handler which sends an email to the member thanking them for paying their fee.
> 
> I'm not really familiar with events in "The Grand Scheme of Things Called DDD", so I'll have to look into that.

Sure, let me know if you'd like to pair program on this or if you would rather have me do it

**Review by vyruz1986 — APPROVED** [review]

**Pull request merged**